frontends/p4/uniqueNames.h

```diff
@@ -114,10 +114,31 @@ class FindSymbols : public Inspector {
 };
 
 class RenameSymbols : public Transform {
+ protected:
     ReferenceMap *refMap;
     RenameMap *renameMap;
 
+    /// Get new name of the current declaration or nullptr if the declaration is not to be renamed.
     IR::ID *getName() const;
+    /// Get new name of the given declaration or nullptr if the declaration is not to be renamed.
+    /// @param decl Declaration *in the original/non-transformed* P4 IR.
+    IR::ID *getName(const IR::IDeclaration *decl) const;
+
+    /// Add a @name annotation ONLY if it does not already exist.
+    /// Otherwise do nothing.
+    static const IR::Annotations *addNameAnnotation(cstring name, const IR::Annotations *annos);
+
+    /// Rename any declaration where we want to add @name annotation with the original name.
+    /// Has to be a template as there is no common base for declarations with annotations member.
+    template <typename D>
+    const IR::Node *renameDeclWithNameAnno(D *decl) {
```

Comment:
Nitpick: Would call this `renameDeclWithNameAnnotation`, even though it is a bit long. All the other functions spell out the annotation.